### ISSUE TITLE
is operator does not report use of unassigned variable in a conditional method call

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#20908]
**Version Used**: https://github.com/dotnet/roslyn/commit/5fbdd5c567b854e9b5f6f1ea64944373a40efb26

**Steps to Reproduce**:
```cs

static class C 
{
    static void M(out object result)
    {
    	M(result is int); // only produces an error if M is not conditional
        result = null;
    }

    [System.Diagnostics.Conditional("X")] // try commenting this
    static void M(bool b) {}
}
```

**Expected Behavior**:  `error CS0269: Use of unassigned out parameter 'result'`

**Actual Behavior**: No error

**Comment by alrz** [comment]
It seems like it's not just for is operator, could anybody confirm with latest bits?

**Comment by alrz** [comment]
Seems like tryroslyn does not compile with DEBUG.
